### PR TITLE
Refine site build

### PR DESCRIPTION
**Pull request by joshfischer1108**
@nwangtw This updates the scripts to generate the static docs with versioning as well as generating javadocs and pydocs correctly.  Once this is merged I think we should be able to update the official site with this one as a starting point.  

CC @jemantheiy

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -131,8 +131,9 @@ com_crashlytics_export_strings.xml
 
 # generated for the website
 website/node_modules/
+website2/website/venv/
```
Comment (nwangtw) [inline .gitignore]: sort.

File: website2/website/versioned_docs/version-0.20.0-incubating/guides-python-topologies.md [hunk 2]
```diff
@@ -82,9 +82,9 @@ Once you've defined spouts and bolts for a topology, you can then compose the to
         my_bolt = CountBolt.spec(par=3, inputs={spout: Grouping.fields("word")})
     ```
 
-## Defining topologies using the [`TopologyBuilder`](/api/python/topology.m.html#heronpy.topology.TopologyBuilder) class
+## Defining topologies using the `TopologyBuilder class
 
-If you create a Python topology using a [`TopologyBuilder`](/api/python/topology.m.html#heronpy.topology.TopologyBuilder), you need to instantiate a `TopologyBuilder` inside of a standard Python main function, like this:
+If you create a Python topology using a `TopologyBuilder1, you need to instantiate a `TopologyBuilder` inside of a standard Python main function, like this:
```
Comment (nwangtw) [inline website2/website/versioned_docs/version-0.20.0-incubating/guides-python-topologies.md]: typo. s/1/`

File: website2/website/versioned_docs/version-0.20.0-incubating/guides-python-topologies.md [hunk 3]
```diff
@@ -82,9 +82,9 @@ Once you've defined spouts and bolts for a topology, you can then compose the to
         my_bolt = CountBolt.spec(par=3, inputs={spout: Grouping.fields("word")})
     ```
 
-## Defining topologies using the [`TopologyBuilder`](/api/python/topology.m.html#heronpy.topology.TopologyBuilder) class
+## Defining topologies using the `TopologyBuilder class
```
Comment (nwangtw) [inline website2/website/versioned_docs/version-0.20.0-incubating/guides-python-topologies.md]: Missing a ` after TopologyBuilder?

File: website2/docs/guides-python-topologies.md [hunk 4]
```diff
@@ -81,9 +81,9 @@ Once you've defined spouts and bolts for a topology, you can then compose the to
         my_bolt = CountBolt.spec(par=3, inputs={spout: Grouping.fields("word")})
     ```
 
-## Defining topologies using the [`TopologyBuilder`](/api/python/topology.m.html#heronpy.topology.TopologyBuilder) class
+## Defining topologies using the `TopologyBuilder class
```
Comment (nwangtw) [inline website2/docs/guides-python-topologies.md]: missing a ` after TopologyBuilder?

File: website2/docs/guides-python-topologies.md [hunk 5]
```diff
@@ -81,9 +81,9 @@ Once you've defined spouts and bolts for a topology, you can then compose the to
...
 
-If you create a Python topology using a [`TopologyBuilder`](/api/python/topology.m.html#heronpy.topology.TopologyBuilder), you need to instantiate a `TopologyBuilder` inside of a standard Python main function, like this:
+If you create a Python topology using a `TopologyBuilder1, you need to instantiate a `TopologyBuilder` inside of a standard Python main function, like this:
```
Comment (nwangtw) [inline website2/docs/guides-python-topologies.md]: s/`TopologyBuilder1/`TopologyBuilder`

**Comment by joshfischer1108** [comment]
@nwangtw all comments addressed (I think).

**Pull request merged**